app/models/trigger_metadata_event_callback.rb

```diff
@@ -15,12 +18,14 @@ def self.publish
   def environment(model)
     if model.aggregate.respond_to?(:component)
       model.aggregate.component.environment
-    elsif model.respond_to?(:service) && model.respond_to?(:sp_component_id)
-      model.service.sp_component.environment
-    elsif model.respond_to?(:service) && model.respond_to?(:msa_component_id)
-      model.service.msa_component.environment
+    elsif model.respond_to?(:service)
+      if model.service.sp_component_id.present?
+        model.service.sp_component.environment
+      elsif model.service.msa_component_id.present?
+        model.service.msa_component.environment
+      end
     else
-      model.aggregate.environment
+      model.aggregate&.environment
```

Comment:
Please remove &. here

Reply:
It would help to have and **explicit** **else**  for when model is **ChangeServiceEvent** that's where all grief came from the **nil** value